### PR TITLE
added tab navigation for the error list

### PR DESCRIPTION
**Pull request by hristoramirez**
fixes: microsoft/pxt-microbit#3139
tab_nav_2
tab_nav_hc2

### CODE REVIEW

File: theme/errorList.less [hunk 1]
```diff
@@ -7,6 +7,10 @@
     min-height: @errorListHeight;
     border-top: 2px @errorListBorderColor solid;
 
+    *:focus {
+        outline: none;
```
Comment (jwunderl) [inline theme/errorList.less:11]: could we see a gif of how it looks w/o this line? The focus is pretty subtle in non high contrast mode, and blended in with the zebra striping a bit to my eyes

Comment (hristoramirez) [reply theme/errorList.less:11]: Here's what it looks without it The one thing was that the outline was always persistent
tab_nav_outline

Comment (jwunderl) [reply theme/errorList.less:11]: Hm, I suppose that's a bit aggressive when you're clicking on it (though I think I might like it for the errors themselves).

Maybe just a tiny bit darker for focus to differentiate it a bit more? Assuming you can do that without the contrast with the text becoming an issue

File: webapp/src/errorList.tsx [hunk 2]
```diff
@@ -77,15 +75,15 @@ export class ErrorList extends React.Component<ErrorListProps, ErrorListState> {
 
         return (
             <div className={`errorList ${isCollapsed ? 'errorListSummary' : ''}`} hidden={!errorsAvailable}>
-                <div className="errorListHeader" role="button" onClick={this.onCollapseClick}>
+                <div className="errorListHeader" role="button" onClick={this.onCollapseClick} onKeyDown={sui.fireClickOnEnter} tabIndex={0}>
```
Comment (jwunderl) [inline webapp/src/errorList.tsx:78]: Oh woops almost forgot, these need aria labels to indicate what they do for screen readers; each div just needs a short description of what it'll do

```suggestion
                <div className="errorListHeader" role="button" aria-label={lf("{0} error list", isCollapsed ? lf("Expand") : lf("Collapse"))} onClick={this.onCollapseClick} onKeyDown={sui.fireClickOnEnter} tabIndex={0}>
```

(note that I just typed that out in the github comment window, sorry if there's a typo ~)

Comment (hristoramirez) [reply webapp/src/errorList.tsx:78]: Thanks I'll add those on and leave the outline on for the errors

**Comment by hristoramirez** [comment]
@jwunderl for elements that contain text, do I add an aria-label with the same text as contained in the element?

**Comment by jwunderl** [comment]
You don't need one if the text already fully describes what will happen

File: webapp/src/errorList.tsx [hunk 3]
```diff
@@ -66,7 +64,7 @@ export class ErrorList extends React.Component<ErrorListProps, ErrorListState> {
                 errorListContent = (
                     <div className="ui selection list">
                         {(errors).map((e, index) =>
-                        <div className="item" key={errorKey(e)} role="button" onClick={createOnErrorMessageClick(e, index)}>
+                        <div className="item" key={errorKey(e)} role="button" onClick={createOnErrorMessageClick(e, index)} onKeyDown={sui.fireClickOnEnter} tabIndex={0}>
```
Comment (jwunderl) [inline webapp/src/errorList.tsx:67]: Maybe have a label that specifies that this will "go to error ", but thats a bit of a nitpick

**Comment by hristoramirez** [comment]
@abchatra

**Pull request merged**